internal/controlplane/identity_events.go

```diff
@@ -148,24 +148,38 @@ func DeleteUser(ctx context.Context, store db.Store, authzClient authz.Client, u
 	defer store.Rollback(tx)
 	qtx := store.GetQuerierWithTransaction(tx)
 
-	user, err := qtx.GetUserBySubject(ctx, userId)
-	if err != nil {
-		if errors.Is(err, sql.ErrNoRows) {
-			// user is already deleted
-			return nil
-		}
+	_, err = qtx.GetUserBySubject(ctx, userId)
+	// If the user doesn't exist, we still want to clean up any associated data
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
 		return fmt.Errorf("error retrieving user %v", err)
 	}
 
+	projects, err := authzClient.ProjectsForUser(ctx, userId)
```

Comment:
Actually, the order was correct. We want to retrieve the list of projects from the authorization system before deleting it. This way we can subsequently delete the project from our database. Else we won't delete it. I added some comments to explain the logic further.

Reply:
You're right. For some other reason I'm not seeing the projects being cleaned up when I try this locally. Still investigating why.

Reply:
Ah I think it's this:
```
-- name: DeleteProject :many
WITH RECURSIVE get_children AS (
    SELECT id, parent_id FROM projects
    WHERE projects.id = $1 AND projects.parent_id IS NOT NULL

    UNION

    SELECT p.id, p.parent_id FROM projects p
    INNER JOIN get_children gc ON p.parent_id = gc.id
)
DELETE FROM projects
WHERE id IN (SELECT id FROM get_children)
RETURNING id, name, metadata, created_at, updated_at, parent_id;
```

There's an extra check for `AND projects.parent_id IS NOT NULL` which doesn't apply anymore since self enrolled projects don't have a parent anymore